### ISSUE TITLE
KeyCurveName.p_256_k resolves to incorrect data

### ISSUE THREAD

**Issue by tezzigator** [Azure/azure-sdk-for-python#10415]
azure-keyvault-keys==4.1.0

[The SDK docs](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.0.0/azure.keyvault.keys.html#azure.keyvault.keys.KeyCurveName) says this resolves as `p_256_k = 'P-256K'` and the SDK does behave this way, unfortunately however Azure keyvault itself returns `SECP256K1` instead of `P-256K`

**To Reproduce**
I use this to check if the keyvault returned P256 or P256K keys returned from list_properties_of_keys():
`if keydat.crv in [KeyCurveName.p_256, KeyCurveName.p_256_k]` and the P256K key I have never shows up.  However this did work:  `if keydat.crv in [KeyCurveName.p_256, 'SECP256K1']` and returns all my P25 and P256K keys.

**Comment by chlowell** [comment]
Thanks for opening this issue. Here's what's happening:

The latest Key Vault version names this curve "P-256K". The prior version named it "SECP256K1" (it's the same curve). For the sake of compatibility Key Vault accepts either name for new keys and didn't change the name on existing keys when the new version rolled out. You can therefore see either name, or both, in a vault. `KeyClient` passes along whatever value Key Vault gives for a string like this. We use the latest name in `KeyCurveName` to prevent confusion about which should be used going forward. I'm sorry this complicates your filtering scenario.

In short, `list_properties_of_keys` returns the (properties of the) key you're looking for but in this case filtering by curve requires considering both possible names.

**Comment by chlowell** [comment]
Thanks again for opening this issue. I'm closing it because the value in `KeyCurveName` is correct for the most recent versions of the Key Vault API and should be used when creating a new key. As I noted above, existing keys may use the prior name for the curve, so no one value can identify all keys with this curve.
